### PR TITLE
Prune inter mode list by SATD

### LINKED ISSUE #1500: Prune inter mode list
Patches like this that add new modes are starting to slow down the encoder a lot:

https://github.com/xiph/rav1e/pull/1496

We should do pruning like we do for intra modes - only try modes with the best SAD score.

### PR DESCRIPTION
**Pull request by takehirokj**
Reduces encoding time by 5.5% to 12.2% at default speed preset in [AWCY](https://beta.arewecompressedyet.com/?job=master%402020-05-08T02%3A52%3A30.506Z&job=prune_inter_num_09_2%402020-05-09T09%3A22%3A24.343Z).

|    PSNR | PSNR Cb | PSNR Cr | PSNR HVS |    SSIM | MS SSIM | CIEDE 2000 |
|    ---: |    ---: |    ---: |     ---: |    ---: |    ---: |       ---: |
| -0.3557 | -0.4733 | -0.4837 |  -0.3510 | -0.3247 | -0.3380 |    -0.3872 |

Closes #1500.

### CODE REVIEW

File: src/rdo.rs [hunk 1]
```diff
@@ -1130,28 +1133,76 @@ fn inter_frame_rdo_mode_decision<T: Pixel>(
...
+      fi.sequence.bit_depth,
+      fi.cpu_feature_level,
+    );
+    satds.push(satd);
   });
 
+  let num_modes_rdo = if fi.config.speed_settings.prediction_modes
+    >= PredictionModesSetting::ComplexAll
+  {
+    inter_mode_set.len()
```
Comment (shssoichiro) [inline src/rdo.rs:1173]: We probably don't need to perform the predict and satd computations for the complex case, correct?

Comment (takehirokj) [reply src/rdo.rs:1173]: Thanks. It's correct.
I will update to skip these for the complex case.
I will also update to skip the sort by satds for the complex case.

**Comment by barrbrain** [comment]
> PSNR 	PSNR Cb 	PSNR Cr 	PSNR HVS 	SSIM 	MS SSIM 	CIEDE 2000
> -0.3557 	-0.4733 	-0.4837 	-0.3510 	-0.3247 	-0.3380 	-0.3872

The comparison was reversed, the BD-rates are positive which is what we would expect from restricting the search.

**Comment by takehirokj** [comment]
I somehow copied reversed one. The correct one is below:
|   PSNR | PSNR Cb | PSNR Cr | PSNR HVS |   SSIM | MS SSIM | CIEDE 2000 |
|   ---: |    ---: |    ---: |     ---: |   ---: |    ---: |       ---: |
| 0.3585 |  0.4772 |  0.4880 |   0.3538 | 0.3271 |  0.3406 |     0.3904 |

**Pull request merged**